naga/src/back/msl/writer.rs

```diff
@@ -1493,13 +1493,21 @@ impl<W: Write> Writer<W> {
                 write!(self.out, "{value}u")?;
             }
             crate::Literal::I32(value) => {
-                write!(self.out, "{value}")?;
+                if value == i32::MIN {
+                    write!(self.out, "({} - 1)", value + 1)?;
+                } else {
+                    write!(self.out, "{value}")?;
+                }
             }
             crate::Literal::U64(value) => {
                 write!(self.out, "{value}uL")?;
             }
             crate::Literal::I64(value) => {
-                write!(self.out, "{value}L")?;
+                if value == i64::MIN {
```

Comment:
Sweet, this is exactly the fix I had in mind, too.

I think we should add a comment explaining exactly what's going on, like we have for the WGSL backend. It's a slightly different scenario though, so I can write that.